src/pyhf/writexml.py

```diff
@@ -48,11 +46,9 @@ def _make_hist_name(channel, sample, modifier='', prefix='hist', suffix=''):
 
 
 def _export_root_histogram(histname, data):
-    hist = TH1.from_numpy((np.asarray(data), np.arange(len(data) + 1)))
-    hist._fName = histname
     if histname in _ROOT_DATA_FILE:
         raise KeyError(f"Duplicate key {histname} being written.")
-    _ROOT_DATA_FILE[histname] = hist
+    _ROOT_DATA_FILE[histname] = (np.asarray(data), np.arange(len(data) + 1))
```

Comment:
```suggestion
    _ROOT_DATA_FILE[histname] = uproot.to_writable(
        (np.asarray(data), np.arange(len(data) + 1))
    )
```

@jpivarski given your other comment, this seems better then directly writing, correct? As if this was ever messed up then we'd get a `TypeError` raised.

Reply:
Just

```python
writable = uproot.to_writable((np.asarray(data), np.arange(len(data) + 1)))
```

by itself ensures that the data passed to `to_writable` will be writable. It will raise a TypeError if it is not recognized as a writable Python type.

The next step of assigning to the file,

```python
_ROOT_DATA_FILE[histname] = writable
```

actually writes it to the file. The disk I/O operations are only invoked when this assignment happens, not when the `to_writable` function is called.

Reply:
> The disk I/O operations are only invoked when this assignment happens, not when the `to_writable` function is called.


Right. Okay, then it seems what I have here (now) is doing as expected. :+1: